app/src/main/java/com/pixlee/pixleeandroidsdk/ui/uioptions/ProductViewFragment.kt

```diff
@@ -0,0 +1,310 @@
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Assert
+
+/**
+ * This shows how you can load photos of Pixlee using PXLAlbum.java
+ */
+class ProductViewFragment : BaseFragment() {
+    override fun getTitleResource(): Int {
+        return R.string.title_java_album
+    }
+
+    private var photoList: ArrayList<PXLPhoto>? = null
+    private var lastImg = R.drawable.grid_2x
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setHasOptionsMenu(true)
+        photoList = ArrayList()
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_product_view, container, false)
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+
+        btnProductView.setOnClickListener {
+            openViwerActivity()
+        }
+
+        lifecycleScope.launch {
+            withContext(Dispatchers.IO) {
+                setupMockedWebServer()
+                intMockServer(getJson("pxl_product.json"))
+            }
+
+            val result = album.getFirstPage()
+            val products = result.photos.firstOrNull()?.products ?: emptyList()
+            rvOption1.adapter = makeProducts(ProductViewHolder.DiscountLayout.CROSS_THROUGH, products)
+            rvOption2.adapter = makeProducts(ProductViewHolder.DiscountLayout.WAS_OLD_PRICE, products)
+            rvOption3.adapter = makeProducts(ProductViewHolder.DiscountLayout.WITH_DISCOUNT_LABEL, products)
+            rvOption4.adapter = makeProducts(null, products)
+            rvOption1Icon.adapter = makeProducts(ProductViewHolder.DiscountLayout.CROSS_THROUGH, products, showShopIcon = true)
+            rvOption2Icon.adapter = makeProducts(ProductViewHolder.DiscountLayout.WAS_OLD_PRICE, products, showShopIcon = true)
+            rvOption3Icon.adapter = makeProducts(ProductViewHolder.DiscountLayout.WITH_DISCOUNT_LABEL, products, showShopIcon = true)
+            rvOption4Icon.adapter = makeProducts(null, products, showShopIcon = true)
+            rvOption1All.adapter = makeProducts(ProductViewHolder.DiscountLayout.CROSS_THROUGH, products, showBookmarkIcon = true, showShopIcon = true)
+            rvOption2All.adapter = makeProducts(ProductViewHolder.DiscountLayout.WAS_OLD_PRICE, products, showBookmarkIcon = true, showShopIcon = true)
+            rvOption3All.adapter = makeProducts(ProductViewHolder.DiscountLayout.WITH_DISCOUNT_LABEL, products, showBookmarkIcon = true, showShopIcon = true)
+            rvOption4All.adapter = makeProducts(null, products, showBookmarkIcon = true, showShopIcon = true)
+        }
+    }
+
+    val album by lazy {
+        PXLKtxAlbum(basicDS, analyticsDS).apply {
+            params = PXLKtxBaseAlbum.Params(
+                    // album images
+                    perPage = 15,
```

Comment:
is this adjustible?

Reply:
yes! they are all adjustable

- val searchId: SearchId,
  ```kotlin
   sealed class SearchId {
        class Album(val id: String) : SearchId()
        class Product(val sku: String) : SearchId()
    }
  ``` 
- var perPage: Int = 30,
- var filterOptions: PXLAlbumFilterOptions? = null,
- var sortOptions: PXLAlbumSortOptions? = null